Modules/Office365DSC/DSCResources/MSFT_SPOSiteGroup/MSFT_SPOSiteGroup.psm1

```diff
@@ -46,52 +49,60 @@ function Get-TargetResource
         Ensure             = "Absent"
     }
 
-    Test-MSCloudLogin -CloudCredential $GlobalAdminAccount `
-        -Platform PnP
 
-    #checking if the site actually exists
-    try
+    if ($RawInputObject)
     {
-        $site = Get-PnPTenantSite $Url
+        $siteGroup = $RawInputObject.SiteGroup
+        $sitePermissions = $RawInputObject.SitePermissions
     }
-    catch
-    {
-        $Message = "The specified site collection doesn't exist."
-        New-Office365DSCLogEntry -Error $_ -Message $Message
-        throw $Message
-        return $nullReturn
-    }
-    try
+    else
     {
         Test-MSCloudLogin -CloudCredential $GlobalAdminAccount `
-            -Platform PnP `
-            -ConnectionUrl $Url
-        $siteGroup = Get-PnPGroup -Identity $Identity
-    }
-    catch
-    {
-        if ($Error[0].Exception.Message -eq "Group cannot be found.")
+        -Platform PnP
+        #checking if the site actually exists
+        try
         {
-            write-verbose -Message "Site group $($Identity) could not be found on site $($Url)"
-
+            $site = Get-PnPTenantSite $Url
         }
-    }
-    if ($null -eq $siteGroup)
-    {
-        return $nullReturn
-    }
+        catch
+        {
+            $Message = "The specified site collection doesn't exist."
+            New-Office365DSCLogEntry -Error $_ -Message $Message
+            throw $Message
+            return $nullReturn
```

Comment:
I think this could never get executed after throwing an exception.

Reply:
Fix'd